src/trait_/eq.rs

```diff
@@ -13,4 +16,38 @@ impl TraitImpl for Eq {
 	fn default_derive_trait(&self) -> DeriveTrait {
 		DeriveTrait::Eq
 	}
+
+	fn supports_skip(&self) -> bool {
+		true
+	}
+
+	fn build_signature(
+		&self,
+		_item: &Item,
+		_traits: &[DeriveTrait],
+		_trait_: &DeriveTrait,
+		body: &TokenStream,
+	) -> TokenStream {
+		quote! {
+			#[inline]
+			fn assert_receiver_is_total_eq(&self) {
+				struct __AssertEq<__T: ::core::cmp::Eq + ?::core::marker::Sized>(::core::marker::PhantomData<__T>);
```

Comment:
technically we could also have our own global `__AssertEq` lying around and use it, but that would then depend on being able to know our path.

Reply:
You can't export types from a proc-macro crate sadly, the only way we could do that is if we make another non-proc-macro crate that holds these types and re-exports the proc-macro from another crate. Basically we would have two crates: `derive-where` and `derive-where-macro`. Should we do that?

Reply:
oh right. If this would be the only reason to do so, I would say no. If we were to add a wrapper crate for some reason, we can do this.